bentoml/server/__init__.py

```diff
@@ -28,7 +34,21 @@ def start_dev_server(
     mb_max_latency: Optional[int] = None,
     run_with_ngrok: Optional[bool] = None,
     enable_swagger: Optional[bool] = None,
+    timeout: Optional[int] = None,
 ):
+    BentoMLContainer.bundle_path.set(bundle_path)
+
+    bento_server = BentoMLContainer.config.bento_server
+    bento_server.port.set(port or skip)
+    bento_server.timeout.set(timeout or skip)
+    bento_server.microbatch.timeout.set(timeout or skip)
+    bento_server.swagger.enabled.set(enable_swagger or skip)
+    bento_server.microbatch.max_batch_size.set(mb_max_batch_size or skip)
+    bento_server.microbatch.max_latency.set(mb_max_latency or skip)
+
+    BentoMLContainer.prometheus_lock.get()  # generate lock before fork
+    BentoMLContainer.forward_port.get()  # generate port before fork
```

Comment:
This suggests we need a eager way to invoke singletons. Maybe singletons should by default eagerly created?

Reply:
I don't think so. Alternatively, I'd more like to remove `multiprocessing` and use `subprocessing`, so that we do not need to care about "process safety" issues.